examples/text/news_topic_classification.py

```diff
@@ -0,0 +1,166 @@
+
+    def copy_tensor(src, dst):
+        assert dst.numel() == src.numel()
+        if move_eos_to_beginning:
+            assert src[-1] == eos_idx
+            dst[0] = eos_idx
+            dst[1:] = src[:-1]
+        else:
+            dst.copy_(src)
+
+    for i, v in enumerate(values):
+        copy_tensor(v, res[i][size - len(v):] if left_pad else res[i][:len(v)])
+    return res
+
+
+def compute_loss(task, roberta, device, learner, loss_func, batch=15):
+    loss = 0.0
+    acc = 0.0
+    for i, (x, y) in enumerate(torch.utils.data.DataLoader(
+            task, batch_size=batch, shuffle=True, num_workers=0)):
+        # RoBERTa ENCODING
+        x = collate_tokens([roberta.encode(sent) for sent in x], pad_idx=1)
+        with torch.no_grad():
+            x = roberta.extract_features(x)
+        x = x[:, 0, :]
+
+        # Moving to device
+        x, y = x.to(device), y.view(-1).to(device)
+
+        output = learner(x)
+        curr_loss = loss_func(output, y)
+        acc += accuracy(output, y)
+        loss += curr_loss / len(task)
+    loss /= len(task)
+    return loss, acc
+
+
+def main(lr=0.005, maml_lr=0.01, iterations=1000, ways=5, shots=1, tps=32, fas=5, device=torch.device("cpu"),
+         download_location="/tmp/text"):
+    text_train = l2l.data.NewsClassification(root=download_location, download=True)
+    train_gen = l2l.data.TaskGenerator(text_train, ways=ways)
+
+    torch.hub.set_dir(download_location)
+    roberta = torch.hub.load('pytorch/fairseq', 'roberta.base')
+    roberta.eval()
+    roberta.to(device)
+    model = Net(num_classes=ways)
+    model.to(device)
+    meta_model = l2l.MAML(model, lr=maml_lr)
+    opt = optim.Adam(meta_model.parameters(), lr=lr)
+    loss_func = nn.NLLLoss(reduction="sum")
+
+    tqdm_bar = tqdm(range(iterations))
+    for iteration in tqdm_bar:
+        iteration_error = 0.0
+        iteration_acc = 0.0
+        for _ in range(tps):
+            learner = meta_model.clone()
+            train_task = train_gen.sample(shots=shots)
+            valid_task = train_gen.sample(shots=shots, classes_to_sample=train_task.sampled_classes)
+
+            # Fast Adaptation
+            for step in range(fas):
+                train_error, _ = compute_loss(train_task, roberta, device, learner, loss_func, batch=shots * ways)
+                learner.adapt(train_error)
+
+            # Compute validation loss
+            valid_error, valid_acc = compute_loss(valid_task, roberta, device, learner, loss_func,
+                                                  batch=shots * ways)
+            iteration_error += valid_error
+            iteration_acc += valid_acc
+
+        iteration_error /= tps
+        iteration_acc /= tps
+        tqdm_bar.set_description("Loss : {:.3f} Acc : {:.3f}".format(iteration_error.item(), iteration_acc))
+
+        # Take the meta-learning step
+        opt.zero_grad()
+        iteration_error.backward()
+        opt.step()
+
+
+if __name__ == '__main__':
```

Comment:
I know everywhere people use argparse in the research community but I really like Google's fire library for doing the exact same thing, and it is so much more intuitive and less verbose.

Reply:
I am open to it. In fact if you want to lead that you probably can, in a follow up PR?

I'd be happy to review it.

Reply:
Yeah! Very familiar with it. Will do it in a follow up PR.